crypto/test/Test/Pos/Crypto/Json.hs

```diff
@@ -35,5 +36,20 @@ golden_ProtocolMagic2AesonDec =
         exampleProtocolMagic2
             "test/golden/json/ProtocolMagic2_Legacy_HasNetworkMagic"
 
+-- Legacy JSON encoding where requiresNetworkMagic was
+-- encoded as "NMMustBeNothing" or "NMMustBeJust"
+
+golden_ProtocolMagic3AesonDec_NMMustBeJust :: Property
+golden_ProtocolMagic3AesonDec_NMMustBeJust =
+    goldenTestJSONDec
+        exampleProtocolMagic3
+            "test/golden/json/ProtocolMagic_Legacy_NMMustBeJust"
+
+golden_ProtocolMagic4AesonDec_NMMustBeNothing :: Property
+golden_ProtocolMagic4AesonDec_NMMustBeNothing =
+    goldenTestJSONDec
+        exampleProtocolMagic4
+            "test/golden/json/ProtocolMagic_Legacy_NMMustBeNothing"
```

Comment:
Do we have equivalent legacy-compatibility tests for `Configuration`? Those are more important than these above 2, because ProtocolMagic is never actually encoded as an Object (only as a Number), whereas the RequiresNetworkMagic field is encoded in `Configuration`'s serialization.

Reply:
No, we don't. Will add some.